impeller/core/resource_binder.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "flutter/fml/logging.h"
+#include "flutter/fml/macros.h"
+#include "impeller/core/buffer_view.h"
+#include "impeller/core/formats.h"
+#include "impeller/core/sampler.h"
+#include "impeller/core/shader_types.h"
+#include "impeller/core/texture.h"
+
+namespace impeller {
+
+//------------------------------------------------------------------------------
+/// @brief      An interface for binding resources. This is implemented by
+///             |Command| and |ComputeCommand| to make CPU resources available
+///             to the GPU.
+///
+struct ResourceBinder {
+  virtual ~ResourceBinder() = default;
```

Comment:
I'm not strongly opposed to adding this pure virtual interface, but it may cause some confusion/friction down the road for whoever ends up adding the next command type.

We're sort of lucky that the bindings match up between raster and compute commands right now, but if we add support for ray trace commands, the bindings won't perfectly match up anymore, and so we'll probably wind up needing to refactor anything that comes to rely on this interface.

Reply:
I think in that case we'd make a `RayTraceCommand` _not_ implement `ResourceBinder` and instead implement some other type from core.

Reply:
Alternatively, in that scenario we could make `ResourceBinder` not be pure virtual.

Reply:
I'm not sure I understand. What is the purpose of `ResourceBinder`? If it's not being used as a generic interface for commands, what is it useful for?

Reply:
https://github.com/flutter/engine/pull/40812#issuecomment-1491174918